AK/StringUtils.cpp

```diff
@@ -102,17 +104,35 @@ Optional<T> convert_to_int(StringView str, TrimWhitespace trim_whitespace)
             sign = -1;
     }
 
+    while (characters[i] == '0' && i < string.length())
+        i++;
+
+    if ((string.length() - i) > NumericLimits<T>::max_digits())
+        return {};
+
+    may_overflow = (string.length() - i) == NumericLimits<T>::max_digits();
+
     T value = 0;
-    for (; i < string.length(); i++) {
+    for (; i < string.length() - may_overflow; i++) {
         if (characters[i] < '0' || characters[i] > '9')
             return {};
 
-        if (__builtin_mul_overflow(value, 10, &value))
-            return {};
+        value *= 10;
+        value += sign * (characters[i] - '0');
+    }
 
-        if (__builtin_add_overflow(value, sign * (characters[i] - '0'), &value))
+    // If we are at the edge of the representable range, we need to check for overflows
```

Comment:
is the 10% claim true across the board? or only for numbers close to NumericLimits::<T>::max() ?

In one of your other comment replies you mentioned that the may_overflow change cost "significant runtime", is it still a net positive of ~10%?

The benchmark looks like it only verifies large numbers, is there any noticable performance difference for small numbers? < 10, 10 - 100, < 1000? I feel like numbers in that range are more common in our codebase and the types of applications/data they commonly interact with.